runtime/strategies/search-tokens-to-handles.js

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2017 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+import {assert} from '../../platform/assert-web.js';
+import {Strategy} from '../../strategizer/strategizer.js';
+import {Recipe} from '../recipe/recipe.js';
+import {RecipeUtil} from '../recipe/recipe-util.js';
+import {Walker} from '../recipe/walker.js';
+
+export class SearchTokensToHandles extends Strategy {
+  constructor(arc) {
+    super();
+    this._arc = arc;
+  }
+
+  async generate(inputParams) {
+    let arc = this._arc;
+    let findMatchingStores = (token, handle) => {
```

Comment:
add a comment: "Finds stores matching the provided token and compatible with the provided handle's type, which are not already mapped into the provided handle's recipe".

Even better, pass in handle.type and handle.recipe explicitly rather than bundled up in handle.
e.g.
`findMatchingUnmappedStores(token, type, recipe)`

Reply:
added comment. i also use `handle` to count connections directions
`RecipeUtil.directionCounts(handle)`
so can't just pass `type` and `recipe` unfortunately.